Porting/valgrindpp.pl

```diff
@@ -49,7 +50,7 @@
 # (do it early, as it may fail)
 my $fh = \*STDOUT;
 if (exists $opt{'output-file'}) {
-  $fh = new IO::File ">$opt{'output-file'}"
```

Comment:
Changing these to use 3-arg open and eliminating the use of `IO::File` would be better.

Reply:
Why `open` is preferred here?

Reply:
I don't know of any formal guidance on using IO::File over open(), but open has the 3-arg form, so you can 
```
open my $fh, ">", $opt{'output-file'}
```
safely, to get the equivalent from IO::File you need the more C style (see fopen):
```
my $fh = IO::File->new(${'output-file'}, "w")
```
The current code has the same issues as the discouraged 2 arg open, where the type of open is mixed with the filename.

Reply:
> I don't know of any formal guidance on using IO::File over open(), but open has the 3-arg form, so you can
> 
> ```
> open my $fh, ">", $opt{'output-file'}
> ```
> 
> safely, to get the equivalent from IO::File you need the more C style (see fopen):
> 
> ```
> my $fh = IO::File->new(${'output-file'}, "w")
> ```
> 
> The current code has the same issues as the discouraged 2 arg open, where the type of open is mixed with the filename.

Thanks, I didn't know of this difference.

I've been using the native `IO::File->new('FILE', 'w')` or the `Path-Tiny` distribution instead of `open` whenever possible for object oriented-ness and readability.

Force pushed to have the nicer looking letter syntax.